src/sentry/middleware/ratelimit.py

```diff
@@ -46,29 +45,39 @@ def process_view(self, request: Request, view_func, view_args, view_kwargs) -> R
         if rate_limit is None:
             return
 
-        self.rate_limit_metadata = above_rate_limit_check(key, rate_limit)
-        if self.rate_limit_metadata.is_limited:
+        rate_limit_metadata = above_rate_limit_check(key, rate_limit)
+        request.rate_limit_limit = rate_limit_metadata.limit
+        request.rate_limit_current_count = rate_limit_metadata.current
+        request.rate_limit_reset = rate_limit_metadata.reset_time
+
+        if rate_limit_metadata.is_limited:
             request.will_be_rate_limited = True
             enforce_rate_limit = getattr(view_func.view_class, "enforce_rate_limit", False)
             if enforce_rate_limit:
                 return HttpResponse(
                     {
                         "detail": DEFAULT_ERROR_MESSAGE.format(
-                            limit=self.rate_limit_metadata.limit,
-                            window=self.rate_limit_metadata.window,
+                            limit=rate_limit_metadata.limit,
+                            window=rate_limit_metadata.window,
                         )
                     },
                     status=429,
                 )
 
     def process_response(self, request: Request, response: Response) -> Response:
-        if self.rate_limit_metadata is not None:
+        if all(
+            (
+                hasattr(request, "rate_limit_limit"),
+                hasattr(request, "rate_limit_current_count"),
+                hasattr(request, "rate_limit_reset"),
+            )
+        ):
             remaining_count = (
-                self.rate_limit_metadata.limit - self.rate_limit_metadata.current
-                if not self.rate_limit_metadata.is_limited
+                request.rate_limit_limit - request.rate_limit_current_count
+                if request.rate_limit_limit > request.rate_limit_current_count
```

Comment:
make remaining_count a property on the rate limit metadata class that you attach to the request